### PR TITLE
Syntax sugars part 2

### PR DESCRIPTION
**Pull request by tmvkrpxl0**
My friend(@PleahMacaka) has suggested me vector deconstructions that even myself wasn't even aware about their existence.
I also added easy conversion between vectors types. this should be adapted for when 4.x comes out though

### CODE REVIEW

**Comment by tmvkrpxl0** [comment]
I intentionally didn't add Vec3i -> Vec3 conversion as there are multiple "flavors" defined in Vec3 such as `Vec3.atLowerCornerOf(Vec3i)`

**Comment by PleahMaCaka** [comment]
very uwu function

File: kfflib/src/main/kotlin/thedarkcolour/kotlinforforge/forge/vectorutil/Vec2Util.kt [hunk 1]
```diff
@@ -13,3 +13,7 @@ public operator fun Vec2.times(times: Float): Vec2 = scale(times)
 public fun Vec2.clone(): Vec2 {
     return Vec2(x, y)
 }
+
+public operator fun Vec2.component1(): Float = x
```
Comment (thedarkcolour) [inline kfflib/src/main/kotlin/thedarkcolour/kotlinforforge/forge/vectorutil/Vec2Util.kt:17]: i forgot this was a thing but i love it

File: kfflib/src/main/kotlin/thedarkcolour/kotlinforforge/forge/vectorutil/Vector3dUtil.kt [hunk 2]
```diff
@@ -37,3 +40,21 @@ public infix fun Vector3d.cross(other: Vector3d): Vector3d {
 public fun Vector3d.clone(): Vector3d {
     return Vector3d(x, y, z)
 }
+
+public fun Vector3d(other: Vec3): Vector3d {
```
Comment (thedarkcolour) [inline kfflib/src/main/kotlin/thedarkcolour/kotlinforforge/forge/vectorutil/Vector3dUtil.kt]: I think a .toVec3d would be better than a fake constructor

File: kfflib/src/main/kotlin/thedarkcolour/kotlinforforge/forge/vectorutil/Vec3iUtil.kt [hunk 3]
```diff
@@ -11,9 +14,30 @@ public operator fun Vec3i.minus(other: Vec3i): Vec3i = subtract(other)
...
+    val (x, y, z) = this
+
     return x * other.x + y * other.y + z * other.z
 }
 
 public fun Vec3i.clone(): Vec3i {
     return Vec3i(x, y, z)
 }
+
+public fun Vec3i(other: Vec3): Vec3i {
```
Comment (thedarkcolour) [inline kfflib/src/main/kotlin/thedarkcolour/kotlinforforge/forge/vectorutil/Vec3iUtil.kt]: I think a .toVec3d would be better than a fake constructor

File: kfflib/src/main/kotlin/thedarkcolour/kotlinforforge/forge/vectorutil/Vector3fUtil.kt [hunk 4]
```diff
@@ -31,3 +33,17 @@ public operator fun Vector3f.times(times: Float): Vector3f {
 public fun Vector3f.clone(): Vector3f {
     return Vector3f(x(), y(), z())
 }
+
+public fun Vector3f(other: Vector3d): Vector3f {
```
Comment (thedarkcolour) [inline kfflib/src/main/kotlin/thedarkcolour/kotlinforforge/forge/vectorutil/Vector3fUtil.kt]: I think a .toVec3d would be better than a fake constructor

File: kfflib/src/main/kotlin/thedarkcolour/kotlinforforge/forge/vectorutil/Vec3Util.kt [hunk 5]
```diff
@@ -14,3 +15,13 @@ public operator fun Vec3.times(times: Double): Vec3 = scale(times)
 public fun Vec3.clone(): Vec3 {
     return Vec3(x, y, z)
 }
+
+public fun Vec3(other: Vector3d): Vec3 {
+    return Vec3(other.x, other.y, other.z)
```
Comment (thedarkcolour) [inline kfflib/src/main/kotlin/thedarkcolour/kotlinforforge/forge/vectorutil/Vec3Util.kt]: I think a .toVec3 would be better than a fake constructor

**Review by thedarkcolour — CHANGES_REQUESTED** [review]
Change the fake constructors to extension functions like .toVec3d()
Everything else looks good

File: kfflib/src/main/kotlin/thedarkcolour/kotlinforforge/forge/vectorutil/Vec2Util.kt (hunk 1 above)
Comment (tmvkrpxl0) [reply kfflib/src/main/kotlin/thedarkcolour/kotlinforforge/forge/vectorutil/Vec2Util.kt:17]: yep

**Review by thedarkcolour — APPROVED** [review]

**Comment by tmvkrpxl0** [comment]
@thedarkcolour So... if all changes are approved, could you merge this?

**Comment by thedarkcolour** [comment]
I have been busy all week. I can merge it when I get home today, should be back in a couple of hours.

**Pull request merged**